### ISSUE TITLE
config flag shorthand not working with v3

### ISSUE THREAD

**Issue by RGunning** [serverless/serverless#10601]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest version?

- [X] Yes, I'm using the latest version

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

serverless cli config shorthand (-c) is no longer working with v3.

Expected:
`npx serverless package -c serverless.test.ts` should run serverless package with the serverless.test.ts config

Actual:
```
sls package -c serverless.test.ts
Environment: darwin, node 16.13.0, framework 3.0.1 (local), plugin 6.0.0, SDK 4.3.0
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
This command can only be run in a Serverless service directory. Make sure to reference a valid config file in the current working directory if you're using a custom config file
```

However, `sls package --config=serverless.test.ts` works as expected. Bug is just with the shorthand flag

### Service configuration (serverless.yml) content

```yaml
import type { AWS } from '@serverless/typescript';

const serverlessConfiguration: AWS = {
  service: 'test-serverless',
  frameworkVersion: '3',
  provider: {
    name: 'aws',
    runtime: 'nodejs14.x',
    apiGateway: {
      minimumCompressionSize: 1024,
      shouldStartNameWithService: true,
    },
    environment: {
      AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
      NODE_OPTIONS: '--enable-source-maps --stack-trace-limit=1000',
    },
    lambdaHashingVersion: '20201221',
  },
};

module.exports = serverlessConfiguration;
```

### Command name and used flags

yarn sls package -c serverless.test.ts

### Command output

```shell
yarn run v1.22.10
$ /private/tmp/test-serverless/node_modules/.bin/sls package -c serverless.test.ts
Environment: darwin, node 16.13.0, framework 3.0.1 (local), plugin 6.0.0, SDK 4.3.0
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
This command can only be run in a Serverless service directory. Make sure to reference a valid config file in the current working directory if you're using a custom config file
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Environment information

```shell
Framework Core: 3.0.1 (local)
Plugin: 6.0.0
SDK: 4.3.0
```

**Comment by mnapoli** [comment]
Thanks for the report, I can indeed reproduce this bug in v3 with a very simple test case:

```yml
# serverless.test.yml
service: test
provider:
  name: aws
```

```bash
# works
sls package --config=serverless.test.ts

# doesn't work
sls package -c serverless.test.ts
```

**Comment by medikoo** [comment]
@RGunning great thanks for reporting. It'll be fixed with #10607
